hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/ReconRocksDB.java

```diff
@@ -17,79 +17,121 @@
+  private ReconUtils reconUtils;
+  private DBStore dbStore;
 
   @VisibleForTesting
   private static final Logger LOG =
-      LoggerFactory.getLogger(ReconContainerDBProvider.class);
-
-  private OzoneConfiguration configuration;
-  private ReconUtils reconUtils;
+          LoggerFactory.getLogger(ReconRocksDB.class);
 
   @Inject
-  public ReconContainerDBProvider(OzoneConfiguration configuration,
-                                  ReconUtils reconUtils) {
+  ReconRocksDB(OzoneConfiguration configuration, ReconUtils reconUtils) {
     this.configuration = configuration;
     this.reconUtils = reconUtils;
+    this.dbStore = provideReconDB();
   }
 
-  @Override
-  public DBStore get() {
-    DBStore dbStore;
+  public DBStore provideReconDB() {
+    DBStore db;
     File reconDbDir =
-        reconUtils.getReconDbDir(configuration, OZONE_RECON_DB_DIR);
+            reconUtils.getReconDbDir(configuration, OZONE_RECON_DB_DIR);
     File lastKnownContainerKeyDb =
-        reconUtils.getLastKnownDB(reconDbDir, RECON_CONTAINER_KEY_DB);
+            reconUtils.getLastKnownDB(reconDbDir, RECON_CONTAINER_KEY_DB);
     if (lastKnownContainerKeyDb != null) {
-      LOG.info("Last known container-key DB : {}",
-          lastKnownContainerKeyDb.getAbsolutePath());
-      dbStore = initializeDBStore(configuration,
-          lastKnownContainerKeyDb.getName());
+      LOG.info("Last known DB : {}",
+              lastKnownContainerKeyDb.getAbsolutePath());
+      db = initializeDBStore(configuration,
+              lastKnownContainerKeyDb.getName());
     } else {
-      dbStore = getNewDBStore(configuration);
+      db = getNewDBStore(configuration);
     }
-    if (dbStore == null) {
+    if (db == null) {
       throw new ProvisionException("Unable to provide instance of DBStore " +
-          "store.");
+              "store.");
     }
+    return db;
+  }
+
+  public DBStore getDbStore() {
     return dbStore;
   }
 
+  public void reInit() throws IOException {
+    File oldDBLocation = dbStore.getDbLocation();
+    try {
+      dbStore.close();
+    } catch (Exception e) {
+      LOG.warn("Unable to close old Recon container key DB at {}.",
+              dbStore.getDbLocation().getAbsolutePath());
+    }
+    dbStore = getNewDBStore(configuration);
+    LOG.info("Creating new Recon Container DB at {}",
+            dbStore.getDbLocation().getAbsolutePath());
+
+    if (oldDBLocation.exists()) {
+      LOG.info("Cleaning up old Recon Container key DB at {}.",
+              oldDBLocation.getAbsolutePath());
+      FileUtils.deleteDirectory(oldDBLocation);
+    }
+  }
+
+  static void clearTable(Table table) throws IOException {
+    if (table == null) {
+      return;
+    }
+    TableIterator<Object, ? extends KeyValue<Object, Object>>
+            tableIterator = table.iterator();
+    while (tableIterator.hasNext()) {
+      KeyValue<Object, Object> entry = tableIterator.next();
+      table.delete(entry.getKey());
+    }
```

Comment:
We could use constant time drop then create CF with RocksDB:

```
RocksDB db;
db.dropColumnFamily();
db.createColumnFamily();
```

but right now these are not exposed in RDBStore on the Ozone side so maybe later.